content/3. Procedurele informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/1. If-Else.md

```diff
@@ -0,0 +1,30 @@
+---
+title: 1. If-Else
+taxonomie:
+  - IB-19.2.Controlestructuur-Beslissingen-If-Else.PI
+---
+> Meer informatie over [[content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/1. If-Else|If-else]]
+### Stappen in de Systematische Probleem Aanpak
+1. **Doel**: Bepaal de reden voor het gebruik van een *If-Else*-structuur. Bijvoorbeeld: Het programma moet beslissingen kunnen nemen op basis van invoer of andere data.
+
+2. **Subdoelen**: Splits het probleem in subdoelen, zoals:
+   - Het vaststellen van de conditie(s) die getoetst moet worden.
+   - Het formuleren van de actie(s) die gedaan moeten worden op basis van de conditie(s).
+   
+3. **Fasen**:
+   - **Fase 1: Analyse van de condities**  
+     Identificeer de condities waarop de beslissing gebaseerd moet zijn (bijvoorbeeld: leeftijd, score, status).
+
+   - **Fase 2: Bepaal de gewenste acties per conditie**  
+     Beschrijf wat het programma moet doen als een conditie waar is, en wat het moet doen als de conditie niet waar is.
+   
+   - **Fase 3: Bouw de If-Else-structuur op**  
+     Implementeer de controlestructuur in code, volgens het structurele model zoals eerder beschreven.
+   
+   - **Fase 4: Test en valideer**  
+     Controleer of de structuur naar behoren werkt door de condities te testen met verschillende waarden.
+
+#### Vuistregels
+- Vermijd het overmatig gebruik van *If-Else*-structuren door alternatieven te overwegen zoals `else if`-clausules of case-structuren indien beschikbaar.
+- Zorg ervoor dat condities duidelijk en logisch zijn, en vermijd te complexe voorwaarden die moeilijk te begrijpen of te onderhouden zijn.
+- Gebruik haakjes `()` in complexe voorwaarden voor leesbaarheid.
```

Comment:
1) Aan welke fasen hangen deze vuistregels?
-> kwaliteitseisen schrijft voor: "per fase zijn vuistregels opgenomen"

2) Wat betreft complexe voorwaarden, soms heb je daar 'gewoon' mee te dealen. Wat dan?
-> Comments bovenaan je functie (of *desnoods* if-else-structuur) die geheugensteun bieden voor wat de functie in-en-uit-gaat. 
-> Probeer de voorwaarden te versimpelen (zei het door abstractie, opdelen, ...)

3) Als vuistregel zou ik juist het verschil aanduiden tussen wanneer je een if-else gebruikt, of een (of meerdere) else if-clauses. (Zijn cases niet vooral van een switch-statement?). Er zit namelijk verschil in. Bovendien is het lastig te bepalen wanneer iets overmatig veel wordt gebruikt. 

4) Ik zie niet zo snel een verwijzing naar de domeinkennis. Kan dat gewoon een link zijn?